### PR TITLE
Sonarqube

### LINKED ISSUE #64: Add the Sonarqube plugin
Add to the 'Overview' tab in the Entity page

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
## Description

adds sonarqube plugin

## Which issue(s) does this PR fix

Fixes #64 

## PR acceptance criteria

_Put an `x` in the boxes that apply_

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Our current list of plugins within the showcase app include:
 - [GitHub Issues plugin](https://github.com/backstage/backstage/tree/master/plugins/github-issues)
 - [Security Insights plugin](https://github.com/RoadieHQ/roadie-backstage-plugins/tree/main/plugins/frontend/backstage-plugin-security-insights)
 - [Keycloak plugin](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/keycloak-backend)
+- [SonaQube plugin](https://roadie.io/backstage/plugins/sonarqube/)
```
Comment (riginoommen) [inline README.md:30]:
```suggestion
- [SonarQube plugin](https://roadie.io/backstage/plugins/sonarqube/)
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Comment by PatAKnight** [comment]
Here is the [issue](https://github.com/janus-idp/backstage-showcase/issues/64) that we created on SonarQube

Also, it looks like there is a formatting issue that prettier is catching in the CI action. If you run `yarn prettier:write` it should fix it for you

**Comment by christophe-f** [comment]
Also the [getting-started.md](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md) documentation needs to be update with the new information.

File: README.md (hunk 1 above)
Comment (raffaelespazzoli) [reply README.md]: done

**Comment by raffaelespazzoli** [comment]
@christophe-f added the getting started doc

**Comment by raffaelespazzoli** [comment]
@christophe-f @riginoommen can you help me understand what the problem is here? The property seems to be there....

File: app-config.yaml [hunk 2]
```diff
@@ -57,6 +57,14 @@ proxy:
     changeOrigin: true
     secure: true
 
+  '/sonarqube':
+    target: ${SONARQUBE_URL}
+    allowedMethods: ['GET']
+    auth: "${SONARQUBE_TOKEN}:"
```
Comment (riginoommen) [inline app-config.yaml:63]:
```suggestion
    auth: ${SONARQUBE_TOKEN}
```

File: app-config.yaml [hunk 3]
```diff
@@ -57,6 +57,14 @@ proxy:
...
+
+sonarQube:
+  baseUrl: ${SONARQUBE_URL}
```
Comment (riginoommen) [inline app-config.yaml:66]: SONARQUBE_URL base URl does n't need /api extension

**Comment by raffaelespazzoli** [comment]
can we merge this now? @christophe-f @riginoommen

File: app-config.yaml [hunk 4]
```diff
@@ -57,6 +57,18 @@ proxy:
     changeOrigin: true
     secure: true
 
+  '/sonarqube':
+    target: https://sonarcloud.io/api
```
Comment (christophe-f) [inline app-config.yaml:61]: Should use `${SONARQUBE_URL}` instead of the hardcoded url.

File: app-config.yaml [hunk 5]
```diff
@@ -57,6 +57,18 @@ proxy:
...
+
+    allowedMethods: ['GET']
+    # uncomment if using a local instnace of sonarqube
+    # auth: "${SONARQUBE_TOKEN}:"
+    # target: ${SONARQUBE_URL}/api
+
+  # uncomment if using a local instnace of sonarqube
+  # sonarQube:
+  #   baseUrl: ${SONARQUBE_URL}
```
Comment (christophe-f) [inline app-config.yaml:70]: This can be removed as the backend is not installed

File: app-config.yaml [hunk 6]
```diff
@@ -57,6 +57,18 @@ proxy:
     changeOrigin: true
     secure: true
 
+  '/sonarqube':
+    target: https://sonarcloud.io/api
+
+    allowedMethods: ['GET']
+    # uncomment if using a local instnace of sonarqube
+    # auth: "${SONARQUBE_TOKEN}:"
```
Comment (christophe-f) [inline app-config.yaml:65]: The token should be enabled.

File: showcase-docs/getting-started.md [hunk 7]
```diff
@@ -25,6 +25,17 @@ The easiest and fastest method for getting started with the Backstage Showcase a
...
+
+   proxy:
+     '/sonarqube':
+       target: ${SONARQUBE_URL}/api
+       allowedMethods: ['GET']
+       auth: '${SONARQUBE_TOKEN}:'
+
+   # comment this out if using sonarcloud
+   sonarQube:
+     baseUrl: ${SONARQUBE_URL}
```
Comment (christophe-f) [inline showcase-docs/getting-started.md:37]: Do we need both the `target` and the `baseUrl`

**Review by christophe-f — CHANGES_REQUESTED** [review]

File: app-config.yaml (hunk 4 above)
Comment (raffaelespazzoli) [reply app-config.yaml:61]: the e2e fails without a real value, so we left the default one

File: app-config.yaml (hunk 5 above)
Comment (raffaelespazzoli) [reply app-config.yaml:70]: it's needed based on the docs. https://roadie.io/backstage/plugins/sonarqube/#:~:text=sonarQube%3A%0A%20%20baseUrl%3A%20https%3A//your.sonarqube.instance.com

do you want me to remove it?

File: showcase-docs/getting-started.md (hunk 7 above)
Comment (raffaelespazzoli) [reply showcase-docs/getting-started.md:37]: apparently yes: https://roadie.io/backstage/plugins/sonarqube/#:~:text=sonarQube%3A%0A%20%20baseUrl%3A%20https%3A//your.sonarqube.instance.com

File: app-config.yaml (hunk 6 above)
Comment (raffaelespazzoli) [reply app-config.yaml:65]: again e2e test was failing if I remember correctly

**Review by christophe-f — APPROVED** [review]

**Pull request merged**